moped-database/migrations/1718314926409_add_council_districts_searchable/up.sql

```diff
@@ -0,0 +1,235 @@
+        UNION ALL
+        SELECT
+            feature_drawn_points.id,
+            feature_drawn_points.component_id,
+            feature_drawn_points.geography::geometry AS geography,
+            st_exteriorring(st_buffer(feature_drawn_points.geography, 7::double precision)::geometry) AS line_geography,
+            null::integer AS signal_id,
+            null::integer AS length_feet
+        FROM feature_drawn_points
+        WHERE feature_drawn_points.is_deleted = false
+        UNION ALL
+        SELECT
+            feature_drawn_lines.id,
+            feature_drawn_lines.component_id,
+            feature_drawn_lines.geography::geometry AS geography,
+            feature_drawn_lines.geography::geometry AS line_geography,
+            null::integer AS signal_id,
+            feature_drawn_lines.length_feet
+        FROM feature_drawn_lines
+        WHERE feature_drawn_lines.is_deleted = false
+    ) AS feature_union
+    GROUP BY feature_union.component_id
+),
+
+subcomponents AS (
+    SELECT
+        mpcs.project_component_id,
+        string_agg(ms.subcomponent_name, ', '::text) AS subcomponents
+    FROM moped_proj_components_subcomponents AS mpcs
+    LEFT JOIN moped_subcomponents AS ms ON mpcs.subcomponent_id = ms.subcomponent_id
+    WHERE mpcs.is_deleted = false
+    GROUP BY mpcs.project_component_id
+),
+
+component_tags AS (
+    SELECT
+        mpct.project_component_id,
+        string_agg((mct.type || ' - '::text) || mct.name, ', '::text) AS component_tags
+    FROM moped_proj_component_tags AS mpct
+    LEFT JOIN moped_component_tags AS mct ON mpct.component_tag_id = mct.id
+    WHERE mpct.is_deleted = false
+    GROUP BY mpct.project_component_id
+),
+
+related_projects AS (
+    SELECT
+        pmp.project_id,
+        concat_ws(', ', pmp.project_id, string_agg(cmp.project_id::text, ', '::text)) AS related_project_ids_with_self,
+        concat_ws(', ', lpad(pmp.project_id::text, 5, '0'), string_agg(lpad(cmp.project_id::text, 5, '0'), ', '::text)) AS related_project_ids_searchable_with_self
+    FROM moped_project AS pmp
+    LEFT JOIN moped_project AS cmp ON pmp.project_id = cmp.parent_project_id
+    GROUP BY pmp.project_id
+),
+
+min_phase_dates AS (
+    WITH min_dates AS (
+        SELECT
+            mpp.project_id,
+            min(mpp.phase_start) AS min_phase_start,
+            min(mpp.phase_end) AS min_phase_end
+        FROM
+            moped_proj_phases AS mpp
+        LEFT JOIN moped_phases AS mp ON mpp.phase_id = mp.phase_id
+        WHERE
+            mpp.is_phase_end_confirmed = false
+            AND mpp.is_phase_start_confirmed = false
+            AND mpp.is_deleted = false
+            AND mp.phase_name_simple = 'Complete'::text
+        GROUP BY
+            mpp.project_id
+    )
+
+    SELECT
+        min_dates.project_id,
+        least(
+            min_dates.min_phase_start,
+            min_dates.min_phase_end
+        ) AS min_phase_date
```

Comment:
Mike, this SQL looks fantastic. one question—do you think this should be moved up to the project list view to sit alongside the `substantial_completion_date` we keep there? it might be good to have for if/when we want to expose it through the Moped editor UI.

Reply:
i thought about this too and went with the easiest route, but you make a good case for having the two substantial fields equally accessible for future needs. I need to make another update in here before merge so I'll pull this up into the plv!